apps/organizations/models.py

```diff
@@ -335,6 +337,87 @@ def build_notification_message__declined(self):
         return msg
 
 
+class BaseAbstractOrganizationJoinRequest(HtkBaseModel):
+    organization = fk_organization(related_name='join_requests', required=True)
+    user = fk_user(
+        related_name='organization_join_requests',
+    )
+    accepted = models.BooleanField(
+        default=None, null=True
+    )  # True: accepted, False: declined, None: not responded yet
+    timestamp = models.DateTimeField(auto_now_add=True)
+    responded_at = models.DateTimeField(blank=True, null=True, default=None)
+
+    class Meta:
+        abstract = True
+        verbose_name = 'Organization Join Request'
+
+    def __str__(self):
+        value = '{organization_name} - {user} - {status}'.format(
+            organization_name=self.organization.name,
+            user=self.user,
+            status=self.status,
+        )
+        return value
+
+    def json_encode(self) -> Dict[str, Any]:
+        """Returns a dictionary that can be `json.dumps()`-ed as a JSON representation of this object"""
+        value = {
+            'id': self.id,
+            'organization': self.organization.name,
+            'user': self.user.profile.get_full_name() if self.user else None,
+            'email': self.email,
+            'accepted': self.accepted,
+            'requested_at': self.timestamp,
+            'responded_at': self.responded_at,
+            'request_message': self.request_message,
```

Comment:
Should we take this out?

I'm thinking of join requests sort of like "Applications"

e.g., a use case would be for a user to say, `I would like to "apply" to join your school, club, organization, etc.`

I think it makes sense to leave out the message so that subclasses can add additional fields to their forms if they would like, such as requiring the "applicant" to explain their experience and qualifications and why they should be considered for membership.

Reply:
Ah okay, will keep this in mind, thanks.